cpp/cmake_modules/FindGandiva.cmake

```diff
@@ -43,7 +43,7 @@ else()
 
   set(GANDIVA_SEARCH_HEADER_PATHS ${GANDIVA_HOME}/include)
 
-  set(GANDIVA_SEARCH_LIB_PATH "${GANDIVA_HOME}")
+  set(GANDIVA_SEARCH_LIB_PATH "${GANDIVA_HOME}/lib")
```

Comment:
This seemed slightly odd, does this line up with other .cmake files?

Reply:
It does: 
- [FindPlasma](https://github.com/apache/arrow/blob/master/cpp/cmake_modules/FindPlasma.cmake#L49)
- [FindArrow](https://github.com/apache/arrow/blob/master/cpp/cmake_modules/FindArrow.cmake#L56)

Reply:
We don't hard code `lib` as @xhochy says.
We need to fix all `FindArrow.cmake`, `FindPlasma.cmake` and `FindGandiva.cmake`.
But it's out of the scope of pull request. We should work on this later.

Reply:
@kou [FindParquet](https://github.com/apache/arrow/blob/master/cpp/cmake_modules/FindParquet.cmake#L55) as well

Reply:
Oh! Good catch!

Reply:
https://issues.apache.org/jira/browse/ARROW-5021